src/lava/lib/optimization/utils/solver_tuner.py

```diff
@@ -1,76 +1,164 @@
-        # TODO : Check that hyperparameters are arguments for solver
-        best_hyperparameters, best_cost, best_step_to_sol = self._random_search(
-            solver,
-            solver_parameters,
-            stopping_condition)
-        succeeded = stopping_condition(best_cost, best_step_to_sol)
-        return best_hyperparameters, succeeded
-
-    def _random_search(self,
-                       solver,
-                       solver_parameters: ty.Dict,
-                       stopping_condition: ty.Callable[
-                           [float, int], bool] = None
-                       ) -> ty.Union[ty.NoReturn, ty.Dict]:
-
-        best_hyperparameters = None
-        best_cost = float("inf")
-        best_step_to_sol = float("inf")
-        problem = solver.problem
-        params_names = self._params_grid.keys()
-        params_grid = list(
-            it.product(*map(lambda k: self._params_grid[k], params_names)))
-        random.shuffle(params_grid)
-
-        for params in params_grid:
-            hyperparameters = dict(zip(params_names, params))
-            solver_parameters["hyperparameters"] = hyperparameters
-            solution = solver.solve(**solver_parameters)
-            # TODO : Implement logic for CSP problems
+             solver_params: ty.Dict,
+             fitness_fn: ty.Callable[[float, int], float],
+             fitness_target: float = None,
+             ):
+        """
+        Perform random search to optimize solver hyper-parameters based on a
+        fitness function.
+
+        Parameters
+        ----------
+        solver: OptimizationSolver
+            Optimization solver to use for solving the problem.
+        solver_params: ty.Dict
+            Parameters for the solver.
+        fitness_fn: ty.Callable[[float, int], float]
+            Fitness function to evaluate a given set of hyper-parameters,
+            taking as input the current cost and number of steps to solution.
+            This is the function that is maximized by the SolverTuner.
+        fitness_target: float, optional
+            Fitness target to reach. If this is not passed, the full grid is
+            explored before stopping search.
+
+        Returns
+        -------
+        best_hyperparams: ty.Dict
+            Dictionary containing the hyper-parameters with the highest fitness.
+        success: bool
+            Flag signaling if the fitness_target has been reached. If no
+            fitness_target is passed, the flag is True.
+        """
+        # TODO : Check that hyperparams are arguments for solver
+        self._stored_rows = 0
+        if self._store.shape[0] < len(self._search_space):
+            self._store = np.zeros(
+                len(self._search_space), dtype=self._store_dtype)
+
+        best_hyperparams = None
+        best_fitness = -float("inf")
+
+        for params in self._search_space:
+            np.random.seed(self._seed)

```

Comment:
Is it useful to set the seed here again? To me it seems like the shuffling takes place in line 43, and np.random is not being called after pine 93 anymore.

Reply:
My point in doing this was to seed the execution of the OptimizationSolver, not the shuffling. Does it make sense to do it here? Probably the best thing to do would be to add a seed parameter to the OptimizationSolver.solve method.